### PR TITLE
add instancesFactory and instance object

### PR DESCRIPTION
**Pull request by rDr4g0n**

### CODE REVIEW

**Comment by rDr4g0n** [comment]
retest this

File: web/ui/src/Hosts/HostDetailsController.js [hunk 1]
```diff
@@ -41,17 +41,26 @@
...
                     actions: [
                         {
-                            classes: "btn-default",
+                            role: "cancel",
+                            label: "close"
+                        },{
+                            classes: "btn-primary",
+                            label: "refresh",
+                            icon: "glyphicon-repeat",
+                            action: function() {
```
Comment (tgjonestx) [inline web/ui/src/Hosts/HostDetailsController.js]: Is a new 'feature'?

**Comment by tgjonestx** [comment]
:+1:

File: web/ui/src/Hosts/HostDetailsController.js (hunk 1 above)
Comment (rDr4g0n) [inline web/ui/src/Hosts/HostDetailsController.js]: yeah a minor tweak to a feature rye added

**Pull request merged**